pypass/passwordstore.py

```diff
@@ -61,22 +61,44 @@ def __init__(
             path=os.path.join(os.getenv("HOME"), ".password-store"),
             git_dir=None,
     ):
-        self.path = path
+        self.path = os.path.abspath(path)
 
-        # Read the .gpg-id
-        gpg_id_path = os.path.join(path, '.gpg-id')
-        if os.path.isfile(gpg_id_path):
-            with open(gpg_id_path, 'r') as gpg_id_file:
-                self.gpg_id = gpg_id_file.read().strip()
-        else:
-            raise Exception("could not find .gpg-id file")
+        # Check if a main .gpg-id exists
+        self._get_gpg_id(self.path)
 
         # Try to locate the git dir
         git_dir = git_dir or os.path.join(self.path, '.git')
         self.uses_git = os.path.isdir(git_dir)
         if self.uses_git:
             self.git_dir = git_dir
 
+    def _is_valid_store_subpath(self, file_location):
+        child_path = os.path.abspath(file_location)
+
+        try:
+            # Requires at least Python 3.5
+            store_commonpath = os.path.commonpath([self.path])
+            child_commonpath = os.path.commonpath([self.path, child_path])
+            return store_commonpath == child_commonpath
+        except AttributeError:
+            # Pre-3.5 fallback
+            commonprefix = os.path.commonprefix([self.path, child_path])
+            return commonprefix.startswith(self.path)
```

Comment:
Would startswith on the expanded paths be enough?

Reply:
Perhaps. I just personally feel it makes most sense for the fallback to be as similar to the main code as possible.

Reply:
Looking at things like https://stackoverflow.com/a/6803714, it is probably also good practice and shows we've explicitly been watching out for directory traversal attacks.

If @aviau also wants me to change it I will, but otherwise I'll keep it as is.

Reply:
I was just wondering...